modules/azure/main.tf

```diff
@@ -16,17 +21,19 @@ module "virtual_network" {
 }
 
 module "custom_data" {
-  source   = "./custom-data"
-  location = var.location
-  api_key  = var.api_key
-  site     = var.site
+  source    = "./custom-data"
+  location  = var.location
+  api_key   = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.api_key.id})"
```

Comment:
Could we have two dedicated variables just like in the equivalent "user_data" module for AWS:
- variable "api_key"
- variable "api_key_secret_arn"

This could avoid the regexp matching and possible confusion.

Reply:
I was actually trying to avoid plain-text keys completely. I added it back now.

As far as the template is concerned, I think this is a pretty clean way to handle it. [Key Vault references](https://learn.microsoft.com/en-us/azure/app-service/app-service-key-vault-references) are reasonably well documented, and I think someone familiar with Azure would not be confused. (That said, _I_ was confused when I saw them for the first time 😉)